src/main/scala/com/ebiznext/comet/job/index/bqload/BigQuerySparkJob.scala

```diff
@@ -31,14 +31,30 @@ class BigQuerySparkJob(
     extends SparkJob
     with BigQueryJobBase {
 
-  override def withExtraSparkConf(): Map[String, String] =
-    if (cliConfig.writeDisposition == "WRITE_TRUNCATE")
-      cliConfig.options ++ Map(
-        "spark.datasource.bigquery.allowFieldAddition"   -> "false",
-        "spark.datasource.bigquery.allowFieldRelaxation" -> "false"
-      )
-    else
-      cliConfig.options
+  val connectorOptions: Map[String, String] = {
+    val optionsOnTruncate = cliConfig.writeDisposition match {
+      case "WRITE_TRUNCATE" =>
+        cliConfig.options -- List("allowFieldAddition", "allowFieldRelaxation")
```

Comment:
Since now we always update the schema before insertion, we don't really need these options at all.
We can directly remove them, like it is done in `Job#withExtraSparkConf`

Reply:
Removing it in case the user set them

Reply:
Yes, but do we still have to check BQ & writeDisposition anymore ?
```suggestion
  val connectorOptions: Map[String, String] = {
    cliConfig.options -- List("allowFieldAddition", "allowFieldRelaxation")
  }
```